prover/src/constraints/evaluator/logup_gkr.rs

```diff
@@ -192,8 +75,18 @@ where
 /// ...
 /// Therefore, we only compute the non-repeating section of the buffer in each iteration, and index
 /// into it accordingly.
-struct LagrangeKernelTransitionConstraintsDivisor<E: FieldElement> {
-    divisor_evals_inv: Vec<E>,
+///
+/// Note that instead of storing `1 / div` for Lagrange and s-column transition constraints, we
+/// store instead `c / div` where `c` is the constraint composition coefficient associated to
+/// divisor `div`. This reduces the number of multiplications during the computation of
+/// evaluations of the constraint composition polynomial. We call `c / div` constraint evaluation
+/// multipliers or just constraint multipliers.
+pub(crate) struct LogUpGkrConstraintsDivisors<E: FieldElement> {
+    lagrange_transition_multipliers: Vec<E>,
+
+    lagrange_boundary_divisors_inv: Vec<E::BaseField>,
```

Comment:
Why can't we store `lagrange_boundary_multipliers` here as we do for transition constraints? If this results in too much refactoring for this PR, let's create an issue for this.

Reply:
Changed to the same approach with multipliers